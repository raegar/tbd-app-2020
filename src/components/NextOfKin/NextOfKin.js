import React, {useState} from "react";
import TextBox from "../TextBox/TextBox";
import "./NextOfKin.css";
import { Container, Row, Col } from "react-bootstrap";
import UCPButton from "../Buttons/UCPButton";
//import Modal from "../Modal/Modal";
import {isMobile} from 'react-device-detect';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import ProgressBar from "../ProgressBar/ProgressBar";


function NextOfKin({ ...props}) {
	const [kinNameInput, setKinNameInput] = useState("");
	const [kinEmailInput, setKinEmailInput] = useState("");
	const [kinTelInput, setKinTelInput] = useState("");
	const [kinAddressLineOne, setKinAddressLineOne] = useState("");
	const [kinAddressLineTwo, setKinAddressLineTwo] = useState("");
	const [kinTownCity, setKinTownCity] = useState("");
	const [kinRegion, setKinRegion] = useState("");
	const [kinPostcode, setKinPostcode] = useState("");
	const [kinInput, setKinInput]= useState("");
	
	const textboxClassName = isMobile ? "mobiletextbox" : "desktoptextbox";
	//progress bar
	const completionRate = [
		{bgcolor: "#005C6E", completed:30 },
	];
	
	//Saves data collected to do with applicants address
	function saveSelectedData() {
		global.ApplicationFormData.name = kinNameInput;
		global.ApplicationFormData.addressLineOne = kinAddressLineOne;
    	global.ApplicationFormData.addressLineTwo = kinAddressLineTwo;
    	global.ApplicationFormData.city = kinTownCity;
    	global.ApplicationFormData.region = kinRegion;
    	global.ApplicationFormData.postcode = kinPostcode;
		global.ApplicationFormData.email = kinEmailInput;
		global.ApplicationFormData.telephone = kinTelInput;

	
		console.log(global.ApplicationFormData);
	}
	
  	return (
		  <div>
			  <Header/>
			{completionRate.map((item, idx)=>(
				<ProgressBar key={idx} bgcolor={item.bgcolor} completed={item.completed}/>
			))}

  			<Container className="es-form-container">
  			<div className="es-form-background">
  				<h3 className="form-title">Next of kin</h3>
  				<span className="ess-form-left">
  					<TextBox
  						title="Name"
  						placeholder="Enter your name"
  						minLength={5}
  						maxLength={60}
  						id="nameinput"
  						type="text"
  						name="name"
  						className={textboxClassName}
  						onChange={(e) => setKinNameInput(e.target.value)}
  						value={kinNameInput}
							required
  					/>
					  
  					<TextBox
  						title="Email"
  						placeholder="Enter your email"
  						minLength={5}
  						maxLength={60}
  						id="kinEmailinput"
  						type="email"
  						name="kinEmail"
  						className={textboxClassName}
  						onChange={(e) => setKinEmailInput(e.target.value)}
  						value={kinEmailInput}
							required
  					/>
  					<TextBox
  						title="Phone No"
  						placeholder="Enter phone number"
  						minLength={5}
  						maxLength={12}
  						id="kinTelinput"
  						type="tel"
  						name="kTelno"
  						className={textboxClassName}
  						onChange={(e) => setKinTelInput(e.target.value)}
  						value={kinTelInput}
  					/>
					  <TextBox
  						title="Net of kin relationship"
  						placeholder="Enter your next of kins relationship"
  						minLength={5}
  						maxLength={60}
  						id="kinRelationship"
  						type="text"
  						name="kinRelationship"
  						className={textboxClassName}
  						onChange={(e) => setKinInput(e.target.value)}
  						value={kinInput}
							required
  					/>
  				</span>
  				<span className="ess-form-right">
				  <TextBox 
							name="kin_address_line_1"
							title="Kin Address Line 1"
							placeholder="Kin Address Line 1"
							minLength={10}
							maxLength={100}
							size={35}
							id="kin-address-1"
							type="text"
							onChange={(e) => setKinAddressLineOne(e.target.value)}
						/>
						<TextBox 
							name="kin_address_line_2"
							title="Kin Address Line 2"
							placeholder="Kin Address Line 2"
							minLength={10}
							maxLength={100}
							size={35}
							id="kin-address-2"
							type="text"
							onChange={(e) => setKinAddressLineTwo(e.target.value)}
						/>
						<TextBox 
							name="kin_town_city"
							title="Kin Town/City"
							placeholder="Kin Town/City"
							minLength={10}
							maxLength={100}
							size={35}
							id="kin-town-city"
							type="text"
							onChange={(e) => setKinTownCity(e.target.value)}
						/>
						<TextBox 
							name="kin_region"
							title="Kin Region"
							placeholder="Kin Region"
							minLength={3}
							maxLength={30}
							size={25}							
							id="kin-region"
							type="text"
							onChange={(e) => setKinRegion(e.target.value)}
						/>
						<TextBox 
							name="kin_postcode"
							title="Kin Postcode"
							placeholder="Kin Postcode"							
							minLength={6}
							maxLength={9}
							size={10}
							id="kin-postcode"
							type="text"
							onChange={(e) => setKinPostcode(e.target.value)}
						/>
  				</span>
  			</div>
  		</Container>
		  <Container>
				<Row id="buttonRow">
					<Col className="centered-buttons">
						<UCPButton to='/EssentialApplicantInfo'
							primary="True"
							className="mediumbutton"
							buttonText="Go Back"
						/>
					</Col>
					<Col className="centered-buttons" onClick={saveSelectedData}>
						<div id="confirmButton">
							<UCPButton
								to= '/AdditionalInformation'
								primary="True"
								className="mediumbutton"
								buttonText="Confirm and Go"
							/>
						</div>
					</Col>
				</Row>
			</Container>
			<Footer/>
		  </div>
  	);
}

export default NextOfKin;