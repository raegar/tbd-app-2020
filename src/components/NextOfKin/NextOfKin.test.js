import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import NextOfKin from './NextOfKin';
import { BrowserRouter as Router} from 'react-router-dom';

//cleans up 
afterEach(cleanup);

it("Renders LevelThreeQualifications", () => {
    const { asFragment } = render(<Router><NextOfKin/></Router>);
    expect(asFragment()).toMatchSnapshot();
});

/*it("Disclaimer Test Id", () => {
    const { getByTestId } = render(<Disclaimer/>);
    expect(getByTestId('ucpDisclaimer'));
});*/
