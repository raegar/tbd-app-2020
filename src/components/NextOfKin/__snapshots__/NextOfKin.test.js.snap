// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NextOfKin 1`] = `
<DocumentFragment>
  <div>
    <div>
      <header
        class="Header"
      >
        <a
          href="/tbd-app-2020"
        >
          <img
            alt="UCP official logo"
            class="UCPLogoBlue"
            data-testid="ucpLogoHeader"
            src="Logo_White.png"
          />
        </a>
      </header>
      <div
        class="Line"
      />
    </div>
    <div
      class="row"
    >
      <div
        style="height: 30px; width: 90%; background-color: rgb(178, 221, 223); border-radius: 50px; margin: 50px;"
      >
        <div
          style="height: 100%; width: 30%; background-color: rgb(0, 92, 110); border-radius: inherit; text-align: right;"
        >
          <span
            style="padding: 5px; color: rgb(255, 255, 255); font-weight: bold;"
          >
             30% 
          </span>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="courseInfoBox undefined"
        >
          <div
            class="courseTitleRow row"
          >
            <h1
              class="courseInfoTitle"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment-dots fa-w-16 icon"
                data-icon="comment-dots"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
                Course Info
            </h1>
          </div>
          <div
            class="row"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              type="button"
            >
              Move UCAS Calculator to Here​
            </button>
          </div>
          <div
            class="uniPageLinkRow row"
          >
            <button
              class="smallbutton"
              data-testid="ucpButton"
              type="button"
            >
              Close Embed​
            </button>
            <div
              class="iframeContainer"
            >
              <iframe
                class="iframeLink"
                src="https://www.ucp.ac.uk/a-z-courses/"
                title="Courses at UCP"
              />
            </div>
          </div>
          <div
            class="uniPageLinkRow row"
          >
            <a
              href="https://www.ucp.ac.uk/a-z-courses/"
            >
              Link to Course information page on UCP website
            </a>
          </div>
          <div
            class="ucasCalculatorRow row"
          >
            <p>
              <b>
                UCAS Calculator:
              </b>
            </p>
            <div
              class="ucasCalculator"
            >
              <div>
                <div
                  class="container"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col"
                    >
                      <select
                        class="qualInputBox"
                        id="qualifications"
                      >
                        <option
                          disabled=""
                        >
                          Please select a qualification...
                        </option>
                        <option
                          class="gradeTypeHeader"
                          disabled=""
                        >
                          A Level and AS
                        </option>
                        <option
                          value="A Level and AS;A Levels"
                        >
                            A Levels
                        </option>
                        <option
                          value="A Level and AS;AS Levels"
                        >
                            AS Levels
                        </option>
                        <option
                          class="gradeTypeHeader"
                          disabled=""
                        >
                          BTEC National (RQF) (From 2016)
                        </option>
                        <option
                          value="BTEC National (RQF) (From 2016);Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)"
                        >
                            Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="mobiletextbox"
                      >
                        <label />
                        <input
                          class="subjectInputBox"
                          maxlength="30"
                          minlength="0"
                          placeholder="Subject (optional)"
                          size="20"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="col"
                    >
                      <br />
                    </div>
                    <div
                      class="col-md-2"
                    >
                      <br />
                      <select
                        class="Dropdown"
                        disabled=""
                      >
                        <option
                          value=""
                        />
                        <option
                          value="none"
                        >
                          ...
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-6"
                    />
                    <div
                      class="col"
                    >
                      <span
                        class="qualPoints"
                        data-testid="qualPoints"
                      >
                        Points: 
                        <b>
                          0
                        </b>
                      </span>
                    </div>
                  </div>
                  <br />
                  <div
                    class="row"
                  >
                    <button
                      class="mediumbutton addQualButton"
                      data-testid="ucpButton"
                      type="button"
                    >
                      Add another qualification​
                    </button>
                  </div>
                  <span
                    class="totalQualPoints"
                  >
                    Total points: 
                    <b
                      data-testid="totalPoints"
                    >
                      0
                    </b>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="es-form-container container"
        >
          <div
            class="es-form-background"
          >
            <h3
              class="form-title"
            >
              Next of kin
            </h3>
            <span
              class="ess-form-left"
            >
              <div
                class="mobiletextbox"
              >
                <label
                  for="nameinput"
                >
                  Name
                </label>
                <input
                  class="desktoptextbox"
                  id="nameinput"
                  maxlength="60"
                  minlength="5"
                  name="name"
                  placeholder="Enter your name"
                  required=""
                  size="20"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kinEmailinput"
                >
                  Email
                </label>
                <input
                  class="desktoptextbox"
                  id="kinEmailinput"
                  maxlength="60"
                  minlength="5"
                  name="kinEmail"
                  placeholder="Enter your email"
                  required=""
                  size="20"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kinTelinput"
                >
                  Phone No
                </label>
                <input
                  class="desktoptextbox"
                  id="kinTelinput"
                  maxlength="12"
                  minlength="5"
                  name="kTelno"
                  placeholder="Enter phone number"
                  size="20"
                  type="tel"
                  value=""
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kinRelationship"
                >
                  Net of kin relationship
                </label>
                <input
                  class="desktoptextbox"
                  id="kinRelationship"
                  maxlength="60"
                  minlength="5"
                  name="kinRelationship"
                  placeholder="Enter your next of kins relationship"
                  required=""
                  size="20"
                  type="text"
                  value=""
                />
              </div>
            </span>
            <span
              class="ess-form-right"
            >
              <div
                class="mobiletextbox"
              >
                <label
                  for="kin-address-1"
                >
                  Kin Address Line 1
                </label>
                <input
                  id="kin-address-1"
                  maxlength="100"
                  minlength="10"
                  name="kin_address_line_1"
                  placeholder="Kin Address Line 1"
                  size="35"
                  type="text"
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kin-address-2"
                >
                  Kin Address Line 2
                </label>
                <input
                  id="kin-address-2"
                  maxlength="100"
                  minlength="10"
                  name="kin_address_line_2"
                  placeholder="Kin Address Line 2"
                  size="35"
                  type="text"
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kin-town-city"
                >
                  Kin Town/City
                </label>
                <input
                  id="kin-town-city"
                  maxlength="100"
                  minlength="10"
                  name="kin_town_city"
                  placeholder="Kin Town/City"
                  size="35"
                  type="text"
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kin-region"
                >
                  Kin Region
                </label>
                <input
                  id="kin-region"
                  maxlength="30"
                  minlength="3"
                  name="kin_region"
                  placeholder="Kin Region"
                  size="25"
                  type="text"
                />
              </div>
              <div
                class="mobiletextbox"
              >
                <label
                  for="kin-postcode"
                >
                  Kin Postcode
                </label>
                <input
                  id="kin-postcode"
                  maxlength="9"
                  minlength="6"
                  name="kin_postcode"
                  placeholder="Kin Postcode"
                  size="10"
                  type="text"
                />
              </div>
            </span>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="row"
            id="buttonRow"
          >
            <div
              class="centered-buttons col"
            >
              <a
                href="/EssentialApplicantInfo"
              >
                <button
                  class="mediumbutton"
                  data-testid="ucpButton"
                  primary="True"
                  type="button"
                >
                  Go Back​
                </button>
              </a>
            </div>
            <div
              class="centered-buttons col"
            >
              <div
                id="confirmButton"
              >
                <a
                  href="/AdditionalInformation"
                >
                  <button
                    class="mediumbutton"
                    data-testid="ucpButton"
                    primary="True"
                    type="button"
                  >
                    Confirm and Go​
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="ucpFooter"
    />
  </div>
</DocumentFragment>
`;
